### PR TITLE
docs(form): fix an incorrect effect in nz-demo-form-dynamic

### LINKED ISSUE #163: An example of the official website is inappropriate
## I'm submitting a...

<pre><code>
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Support request => Please do not submit support request here
</code></pre>

## Current behavior

I have 5 items (0,1,2,3,4).
Touch the delete button of the item named 0,so the first item has gone 
Next ,Touch the delete button of the item named 1,but * the item named 2* has gone
## Expected behavior

Touch the delete button of the item named 1,* the item named 1* has gone

## Minimal reproduction of the problem with instructions

The Index of the array has refreshed.But the item didn't.
## What is the motivation / use case for changing the behavior?

More precise

## Environment

<pre><code>
Angular version: 4.4.3

ng-zorro-antd version: 0.4.0

Browser:
- [x] Chrome (desktop) version 60.0.3112.78 (Official Build) (64-bit)
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] IE version XX

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by thegatheringstorm**
close #163

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #163 

## What is the new behavior?
Fix the bug described in #163 

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

#### the element in 'controlArray' (NzDemoFormDynamicComponent) has been changed to the object like
```js
{
"id":1,
"controlInstance":"passenger1"
}
```

### CODE REVIEW

**Comment by thegatheringstorm** [comment]
I'm sorry to say that I had to open a new PR for I delete my Repo for the old PR.

File: src/showcase/nz-demo-form/nz-demo-form-dynamic.component.ts [hunk 1]
```diff
@@ -53,32 +61,39 @@ import {
...
-    this.validateForm.addControl(`passenger${this.index}`, new FormControl(null, Validators.required));
-    this.index++;
+    const id = (this.controlArray.length > 0) ? this.controlArray[this.controlArray.length - 1].id + 1 : 0;
+
+    const control = {
+      id,
+      controlInstance: `passenger${id}`
+    };
+    const index = this.controlArray.push(control);
+    console.log(this.controlArray[this.controlArray.length - 1]);
```
Comment (SangKa) [inline src/showcase/nz-demo-form/nz-demo-form-dynamic.component.ts]: `console.log` should be removed.

File: src/showcase/nz-demo-form/nz-demo-form-dynamic.component.ts [hunk 2]
```diff
@@ -53,32 +61,39 @@ import {
...
   }
 
   removeField(i, e: MouseEvent) {
     e.preventDefault();
     if (this.controlArray.length > 1) {
-      this.controlArray.splice(i, 1);
-      this.validateForm.removeControl(`passenger${this.index}`);
+      const index = this.controlArray.indexOf(i);
+      this.controlArray.splice(index, 1);
+      console.log(this.controlArray);
```
Comment (SangKa) [inline src/showcase/nz-demo-form/nz-demo-form-dynamic.component.ts]: `console.log` should be removed.

Comment (vthinkxie) [reply src/showcase/nz-demo-form/nz-demo-form-dynamic.component.ts]: console is fine in doc component

**Pull request merged**